libgalois/src/analytics/Utils.cpp

```diff
@@ -19,15 +19,15 @@
 
 #include "katana/analytics/Utils.h"
 
-#include <katana/Random.h>
+#include "katana/Random.h"
 
 uint32_t
 katana::analytics::SourcePicker::PickNext() {
   uint32_t source;
   do {
     source = RandomUniformInt(graph.size());
-  } while (
-      std::distance(graph.edges(source).begin(), graph.edges(source).end()));
+  } while (graph.edges(source).empty());
+  KATANA_LOG_ASSERT(graph.edges(source));
```

Comment:
Confused by this. `edges()` returns a range, so what is this asserting?

Also, I think `assert()` (debug-only) is better here than `KATANA_LOG_ASSERT` (which is also invoked in release builds).

Reply:
C++ ranges are "true-ish" if they are not empty. So this is equivalent to `graph.edges(source).size() > 0`.

Yea. Debug only would be good. But I'd like to have a KATANA macro for it. The reason is that the aborting KATANA macros will eventually actually throw an exception to abort the C++ code in a way that doesn't crash the Python interpreter. Python extensions are not supposed to abort. I'll post about this on code-related.

Reply:
Thanks